flake-module.nix

```diff
@@ -203,13 +224,25 @@ in
       in
       {
         packages =
-          lib.mapAttrs
-            (_: project: project.package)
-            projects;
-        apps =
-          lib.mapAttrs
-            (_: project: project.app)
-            projects;
+          lib.foldl (x: y: x // y) ({ }) (
+            lib.mapAttrsToList
+              (projectName: project:
+                lib.listToAttrs
+                  (lib.mapAttrsToList
+                    (packageName: package: {
+                      name =
+                        # Prefix package names with the project name (unless
+                        # project is named `default`)
+                        if projectName == "default"
+                        then packageName
+                        else "${projectName}-${packageName}";
+                      value = package;
+                    })
+                    project.packages
+                  )
```

Comment:
I wonder if the Nix code here can be refactored?

Reply:
Resolving for now, in the absence of something better; @roberth and others can feel free to reopen if there are any suggestions

Reply:
`listToAttrs` after `mapAttrsToList` should be equal to `mapAttrs'`.

The `foldl` can perhaps be changed to `mkMerge` like below.

Perhaps it could be cut up by naming things.

Reply:
Done